ext/date/php_date.c

```diff
@@ -5976,3 +5972,27 @@ static zval *date_period_get_property_ptr_ptr(zend_object *object, zend_string *
 
 	return zend_std_get_property_ptr_ptr(object, name, type, cache_slot);
 }
+
+static HashTable *date_period_get_properties_for(zend_object *object, zend_prop_purpose purpose)
+{
+	switch (purpose) {
```

Comment:
I'm a bit confused by the implementation: why this switch is necessary here.

Reply:
I copy-pasted this piece of code. I'm not sure why Nikita added the switch there... but after a closer examination, it seemed that `ZEND_PROP_PURPOSE_GET_OBJECT_VARS` is missing from the cases 🤔 I don't see the reason... but in any case, I now removed the switch case so that the custom handler kicks in in all cases now.